packages/teams-js/src/private/externalAppAuthenticationForCEA.ts

```diff
@@ -0,0 +1,269 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { validateId } from '../internal/utils';
+import { AppId } from '../public';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../public/constants';
+import { runtime } from '../public/runtime';
+import { externalAppAuthentication } from './externalAppAuthentication';
+
+const externalAppAuthenticationTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * @hidden
+ * Namespace to delegate authentication requests to the host for custom engine agents
+ * @internal
+ * Limited to Microsoft-internal use
+ * @beta
+ */
+export namespace externalAppAuthenticationForCEA {
+  /**
+   * @beta
+   * @hidden
+   * Signals to the host to perform SSO authentication for the application specified by the app ID, and then send the authResult to the application backend.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId Id of the application backend for which the host should attempt SSO authentication.
+   * @param conversationId ConversationId To tell the bot what conversation the calls are coming from
+   * @param authTokenRequest Parameters for SSO authentication
+   * @throws InvokeError if the host encounters an error while authenticating
+   * @returns A promise that resolves when authentication succeeds and rejects with InvokeError on failure
+   */
+  export async function authenticateWithSSO(
+    appId: AppId,
+    conversationId: string,
+    authTokenRequest: externalAppAuthentication.AuthTokenRequestParameters,
+  ): Promise<void> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    validateId(conversationId, new Error('conversation id is not valid.'));
+
+    const [error] = await sendMessageToParentAsync<[externalAppAuthentication.InvokeError]>(
+      getApiVersionTag(
+        externalAppAuthenticationTelemetryVersionNumber,
+        ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSO,
+      ),
+      ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSO,
+      [appId.toString(), conversationId, authTokenRequest.claims, authTokenRequest.silent],
+    );
+    if (error) {
```

Comment:
>error [](http://example.com/codeflow?start=8&length=5)

Based on the type I see being used in the `sendMessageToParentAsync` call, it seems like `undefined` / `null` is not a value return type -- it must be a `externalAppAuthentication.InvokeError` object. If this hub SDK call can actually return `undefined` / `null`, should the return type of the "send" call actually be `externalAppAuthentication.InvokeError | undefined` ?

Reply:
Added undefined to the type for non error cases